docs/lab/02/02.md

```diff
@@ -0,0 +1,223 @@
+
+Because writing functionalities on an MCU is quite complicated and requires a lot of time using bare metal. Over time, several layers of abstraction were added to the programming languages and frameworks appeared to make the writing of new functionalities more efficient. One of these frameworks is Embassy-rs, which is a level of abstraction added on top of Rust to make the development process more efficient and faster.
+
+**We have to follow several steps to configure a pin in embassy-rs:**
+
+1. We need the specialized library from embassy rs for controlling the GPIO pins and certain components from it.
+
+```rust
+use gpio::{Input, Level, Output, Pull};
+```
+
+2. We have to define an gpio, choose its type and choose the corresponding pin number.
+
+**Output**
+```rust
+let mut gpio_name = Output::new(p.PIN_n, Level::Low);
+```
+
+You must replace `n` in `PIN_n` with the pin number. `Level::Low` sets the initial output of the pin, for example if you connect a led and initially want it to be off, set Level::Low, if you want it to be on, set `Level::High`.
+
+**Input**
+```rust
+let gpio_name = Input::new(p.PIN_n, Pull::Up);
+```
+
+For the input type pins, as well as the output ones, we must set the pin number in `PIN_n`, `n` being the number in the pin diagram. `Pull::Up` or `Pull::Down` activates the internal pull up or pull down resistors of the GPIO pin.
+
+:::warning 
+
+For a correct use of the buttons, use pull-up, pull-down resistors depending on the mode of operation of the button. Check [01 - Hardware Introduction](https://embedded-rust-101.wyliodrin.com/docs/lab/01#buttons)
+
+:::
+
+## Control GPIO Pins
+
+As we saw in the previous chapters, the GPIO pins can be used to communicate with the environment through external elements. Next we will see how we can do this using Embassy-rs.
+
+### Writing Rust code to control LEDs using GPIO pins.
+
+The LEDs can be controlled by gpio because the pins can output electric current depending on their configuration (read the datasheet). Therefore, after you configure the pin to correspond with your led, it is necessary to control the voltage for the respective led, so we have to look at the panel of the GPIO pin and choose the appropriate resistance.
+
+**Steps to turn on/off a led:**
+
+1. First of all, we will define our LED and assign it the Level::Low property to be initially off.
+```rust
+let mut led = Output::new(p.PIN_17, Level::Low);
+```
+
+2. To control the flow of current passing through the pin, we have 2 functions to set the pin to low or high depending on the need.
+```rust
+led.set_high();
+led.set_low();
+```
+
+:::warning 
+
+The GPIO pins output a standard output current depending on the microcontroller. To be sure that the LED will work normally and there is no risk of destruction, it is recommended to calculate and add a resistor to control the current level.
+
+:::
+
+<div align="center">
+![LED GPIO](images/led-gpio17.png)
+</div>
```

Comment:
```suggestion
:::warning 

The maximum current that GPIO pins output depends on the microcontroller. To be sure that the LED will work normally and there is no risk of destruction, a resistor has to be added to limit the current level *below* the maximum GPIO output current.

<div align="center">
![LED GPIO](images/led-gpio17.png)
</div>

:::
```

I would use an image with the Raspberry Pi Pico, this one seems to be for the Raspberry Pi.